[CI] PercolatorFieldMapperTests testCreateCandidateQuery failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/wvies6hsm6qec/tests/:modules:percolator:test/org.elasticsearch.percolator.PercolatorFieldMapperTests/testCreateCandidateQuery

**Reproduction line:**
`./gradlew ':modules:percolator:test' --tests "org.elasticsearch.percolator.PercolatorFieldMapperTests.testCreateCandidateQuery" -Dtests.seed=1374F441FACCB75E -Dtests.locale=zh-HK -Dtests.timezone=Antarctica/Palmer -Druntime.java=11`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.percolator.PercolatorFieldMapperTests&tests.test=testCreateCandidateQuery

**Failure excerpt:**
```
java.lang.AssertionError: (No message provided)

  at __randomizedtesting.SeedInfo.seed([1374F441FACCB75E:EA4B104E7BD5AA45]:0)
  at org.junit.Assert.fail(Assert.java:86)
  at org.junit.Assert.assertTrue(Assert.java:41)
  at org.junit.Assert.assertFalse(Assert.java:64)
  at org.junit.Assert.assertFalse(Assert.java:74)
  at org.elasticsearch.percolator.PercolatorFieldMapperTests.testCreateCandidateQuery(PercolatorFieldMapperTests.java:361)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:566)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:834)

```

--- a/modules/percolator/src/test/java/org/elasticsearch/percolator/CandidateQueryTests.java
+++ b/modules/percolator/src/test/java/org/elasticsearch/percolator/CandidateQueryTests.java
@@ -868,10 +868,13 @@ public class CandidateQueryTests extends ESSingleNodeTestCase {
         }
 
         // This will trigger using the TermsQuery instead of individual term query clauses in the CoveringQuery:
+        int origMaxClauseCount = BooleanQuery.getMaxClauseCount();
         try (Directory directory = new ByteBuffersDirectory()) {
+            final int maxClauseCount = 100;
+            BooleanQuery.setMaxClauseCount(maxClauseCount);
             try (IndexWriter iw = new IndexWriter(directory, newIndexWriterConfig())) {
                 Document document = new Document();
-                for (int i = 0; i < 4096; i++) {
+                for (int i = 0; i < maxClauseCount; i++) {
                     int fieldNumber = 2 + i;
                     document.add(new StringField("field", "value" + fieldNumber, Field.Store.NO));
                 }
@@ -897,6 +900,8 @@ public class CandidateQueryTests extends ESSingleNodeTestCase {
                 assertEquals(1, topDocs.scoreDocs[0].doc);
                 assertEquals(2, topDocs.scoreDocs[1].doc);
             }
+        } finally {
+            BooleanQuery.setMaxClauseCount(origMaxClauseCount);
         }
     }
 

--- a/modules/percolator/src/test/java/org/elasticsearch/percolator/CandidateQueryTests.java
+++ b/modules/percolator/src/test/java/org/elasticsearch/percolator/CandidateQueryTests.java
@@ -807,7 +807,6 @@ public class CandidateQueryTests extends ESSingleNodeTestCase {
         assertEquals(2, topDocs.scoreDocs[1].doc);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/75592")
     public void testPercolateSmallAndLargeDocument() throws Exception {
         List<LuceneDocument> docs = new ArrayList<>();
         BooleanQuery.Builder builder = new BooleanQuery.Builder();
@@ -872,7 +871,7 @@ public class CandidateQueryTests extends ESSingleNodeTestCase {
         try (Directory directory = new ByteBuffersDirectory()) {
             try (IndexWriter iw = new IndexWriter(directory, newIndexWriterConfig())) {
                 Document document = new Document();
-                for (int i = 0; i < 1024; i++) {
+                for (int i = 0; i < 4096; i++) {
                     int fieldNumber = 2 + i;
                     document.add(new StringField("field", "value" + fieldNumber, Field.Store.NO));
                 }

--- a/modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorFieldMapperTests.java
+++ b/modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorFieldMapperTests.java
@@ -337,33 +337,40 @@ public class PercolatorFieldMapperTests extends ESSingleNodeTestCase {
 
 
     public void testCreateCandidateQuery() throws Exception {
-        addQueryFieldMappings();
+        int origMaxClauseCount = BooleanQuery.getMaxClauseCount();
+        try {
+            final int maxClauseCount = 100;
+            BooleanQuery.setMaxClauseCount(maxClauseCount);
+            addQueryFieldMappings();
 
-        MemoryIndex memoryIndex = new MemoryIndex(false);
-        StringBuilder text = new StringBuilder();
-        for (int i = 0; i < 4094; i++) {
-            text.append(i).append(' ');
+            MemoryIndex memoryIndex = new MemoryIndex(false);
+            StringBuilder text = new StringBuilder();
+            for (int i = 0; i < maxClauseCount - 2; i++) {
+                text.append(i).append(' ');
+            }
+            memoryIndex.addField("field1", text.toString(), new WhitespaceAnalyzer());
+            memoryIndex.addField(new LongPoint("field2", 10L), new WhitespaceAnalyzer());
+            IndexReader indexReader = memoryIndex.createSearcher().getIndexReader();
+
+            Tuple<BooleanQuery, Boolean> t = fieldType.createCandidateQuery(indexReader, Version.CURRENT);
+            assertTrue(t.v2());
+            assertEquals(2, t.v1().clauses().size());
+            assertThat(t.v1().clauses().get(0).getQuery(), instanceOf(CoveringQuery.class));
+            assertThat(t.v1().clauses().get(1).getQuery(), instanceOf(TermQuery.class));
+
+            // Now push it over the edge, so that it falls back using TermInSetQuery
+            memoryIndex.addField("field2", "value", new WhitespaceAnalyzer());
+            indexReader = memoryIndex.createSearcher().getIndexReader();
+            t = fieldType.createCandidateQuery(indexReader, Version.CURRENT);
+            assertFalse(t.v2());
+            assertEquals(3, t.v1().clauses().size());
+            TermInSetQuery terms = (TermInSetQuery) t.v1().clauses().get(0).getQuery();
+            assertEquals(maxClauseCount - 1, terms.getTermData().size());
+            assertThat(t.v1().clauses().get(1).getQuery().toString(), containsString(fieldName + ".range_field:<ranges:"));
+            assertThat(t.v1().clauses().get(2).getQuery().toString(), containsString(fieldName + ".extraction_result:failed"));
+        } finally {
+            BooleanQuery.setMaxClauseCount(origMaxClauseCount);
         }
-        memoryIndex.addField("field1", text.toString(), new WhitespaceAnalyzer());
-        memoryIndex.addField(new LongPoint("field2", 10L), new WhitespaceAnalyzer());
-        IndexReader indexReader = memoryIndex.createSearcher().getIndexReader();
-
-        Tuple<BooleanQuery, Boolean> t = fieldType.createCandidateQuery(indexReader, Version.CURRENT);
-        assertTrue(t.v2());
-        assertEquals(2, t.v1().clauses().size());
-        assertThat(t.v1().clauses().get(0).getQuery(), instanceOf(CoveringQuery.class));
-        assertThat(t.v1().clauses().get(1).getQuery(), instanceOf(TermQuery.class));
-
-        // Now push it over the edge, so that it falls back using TermInSetQuery
-        memoryIndex.addField("field2", "value", new WhitespaceAnalyzer());
-        indexReader = memoryIndex.createSearcher().getIndexReader();
-        t = fieldType.createCandidateQuery(indexReader, Version.CURRENT);
-        assertFalse(t.v2());
-        assertEquals(3, t.v1().clauses().size());
-        TermInSetQuery terms = (TermInSetQuery) t.v1().clauses().get(0).getQuery();
-        assertEquals(4095, terms.getTermData().size());
-        assertThat(t.v1().clauses().get(1).getQuery().toString(), containsString(fieldName + ".range_field:<ranges:"));
-        assertThat(t.v1().clauses().get(2).getQuery().toString(), containsString(fieldName + ".extraction_result:failed"));
     }
 
     public void testExtractTermsAndRanges_numberFields() throws Exception {

--- a/modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorFieldMapperTests.java
+++ b/modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorFieldMapperTests.java
@@ -341,7 +341,7 @@ public class PercolatorFieldMapperTests extends ESSingleNodeTestCase {
 
         MemoryIndex memoryIndex = new MemoryIndex(false);
         StringBuilder text = new StringBuilder();
-        for (int i = 0; i < 1022; i++) {
+        for (int i = 0; i < 4094; i++) {
             text.append(i).append(' ');
         }
         memoryIndex.addField("field1", text.toString(), new WhitespaceAnalyzer());
@@ -361,7 +361,7 @@ public class PercolatorFieldMapperTests extends ESSingleNodeTestCase {
         assertFalse(t.v2());
         assertEquals(3, t.v1().clauses().size());
         TermInSetQuery terms = (TermInSetQuery) t.v1().clauses().get(0).getQuery();
-        assertEquals(1023, terms.getTermData().size());
+        assertEquals(4095, terms.getTermData().size());
         assertThat(t.v1().clauses().get(1).getQuery().toString(), containsString(fieldName + ".range_field:<ranges:"));
         assertThat(t.v1().clauses().get(2).getQuery().toString(), containsString(fieldName + ".extraction_result:failed"));
     }

